src/terminal.cpp

```diff
@@ -2785,6 +2781,10 @@ void Terminal::consumeCSI()
   bool questionMarkFound;
   int params[FABGLIB_MAX_CSI_PARAMS];
   int paramsCount;
+
+  // zero the memory allocated to params
+  memset(params, 0, FABGLIB_MAX_CSI_PARAMS * sizeof(int))
```

Comment:
Looks good to me.

Just noticed that Terminal::consumeParamsAndGetCode() is also called by Terminal::consumeDCS(), so probably the same params array initialization should be added there. Although, the params array doesn't seem to be used for anything important in consumeDCS() code, except in some debug logging code at the end of the method.

Reply:
Done.